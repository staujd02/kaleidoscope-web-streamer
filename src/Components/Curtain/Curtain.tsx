import React from 'react';

class Curtain extends React.Component<CurtainProps, CurtainState> {
    render(){
        const { open } = this.props;
        return open ? this.props.children : null;
    }
}

export default Curtain;