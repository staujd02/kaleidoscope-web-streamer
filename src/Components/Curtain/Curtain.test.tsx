import React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import Curtain from './Curtain';

describe("The Curtain", () => {

    const children = <div>Now you see me!</div>;
    let wrapper: ShallowWrapper<CurtainProps, CurtainState>;

    beforeEach(() => {
        wrapper = shallow(
            <Curtain open={true}>
                {children}
            </Curtain>);
    });

    it('renders correctly', () => expect(wrapper.html()).toMatchSnapshot());

    describe("given the curtain is rolled back", () => {

        beforeEach(() => {
            wrapper.setProps({
                open: true
            });
        });

        it("displays its children", () => {
            expect(wrapper.contains(children)).toEqual(true);
        });
    });
    
    describe("given the curtain is across the stage", () => {

        beforeEach(() => {
            wrapper.setProps({
                open: false
            });
        });

        it("does not displays its children", () => {
            expect(wrapper.contains(children)).toEqual(false);
        });
    });

});