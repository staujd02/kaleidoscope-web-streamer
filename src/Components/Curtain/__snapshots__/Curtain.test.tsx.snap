// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Curtain renders correctly 1`] = `"<div>Now you see me!</div>"`;
